### ISSUE TITLE
Panel: ARIA nested controls warning because header has role button

### ISSUE THREAD

**Issue by djmj** [primefaces/primefaces#14739]
### Describe the bug

Testing current PF showcase https://showcase.primefaces.org/ui/panel/panel.xhtml against ARIA rules at this site:

https://www.accessibilitychecker.org 

it reports:

_Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies_

I think the problem is the `role="button"` on the complete header instead just the small toggle button. This creates nested attributes and the screenreader is seeing anchor controls inside a button.

If this is really the case (i dont know much about ARIA), maybe just remove the role attribute on the panel and put the `aria-controls` attribute on the toggle button instead.

### PrimeFaces edition

None

### PrimeFaces version

15.0.14

### JSF implementation

Mojarra

### JSF version

4.0.12

### Java version

17

**Comment by melloware** [comment]
I think the problem is in the showcase it has `toggleableHeader="true"` so the whole header is clickable button.

**Comment by djmj** [comment]
In my observation at least in PF 14 it also comes without this flag as soon the `toggleable="true"`. In this case i was even more surprise.

People who need screenreaders and ARIA are good enough with the toggle button i think, thats why i think just move those aria properties to the toggle button instead and let it be clickable for the normal people.

**Comment by melloware** [comment]
yeah let me look at Git Blame this does seem odd.

**Comment by melloware** [comment]
original ticket: https://github.com/primefaces/primefaces/issues/9993

@EugenFischer can you weigh in here?

**Comment by melloware** [comment]
`role="button"` was then added in https://github.com/primefaces/primefaces/issues/10653

**Comment by melloware** [comment]
@djmj @EugenFischer i think these different tools, Lighthouse, Axe, https://www.accessibilitychecker.org/ all report different things that "could" be a problem but are not necessarily a problem and the screen readers like NVDA are actually reading them just fine.  The real test is with a ScreenReader like with teh issue @prusuteodora recently reported.

So it would be nice to get actual screenreader data.

**Comment by djmj** [comment]
Yeah, every report is different, so i dont know. 

Not sure about PF 15 but in PF14 it adds `role="button"` to the header, even if header is not clickable. Maybe this should be removed if `toggleableHeader="false"`?

**Comment by melloware** [comment]
yeah waiting for @EugenFischer since he originally requested the way it is now.  But yes th ToggleableHeader makes more sense I agree.
